packages/ckeditor5-enter/src/enterobserver.ts

```diff
@@ -32,14 +35,26 @@ export default class EnterObserver extends Observer {
 		super( view );
 
 		const doc = this.document;
+		let shiftPressed = false;
+
+		doc.on<ViewDocumentKeyDownEvent>( 'keydown', ( evt, data ) => {
```

Comment:
What about `'keyup'`? What if the user had released `shift` before she pressed `enter`?

Reply:
This listener does not care what key was pressed, just the `shiftKey` flag that is provided in every key event. So the shift key state will be from the last `keydown` event and it would be `enter` key that triggered `beforeinput`.